app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/organisation/GrsController.scala

```diff
@@ -161,81 +161,74 @@ class GrsController @Inject() (
   private def saveRegistrationDetails(journeyId: String)(implicit
     hc: HeaderCarrier,
     request: JourneyRequest[AnyContent]
-  ): Future[Either[ServiceError, Registration]] =
-    request.registration.organisationDetails.organisationType match {
+  ): Future[Either[ServiceError, Registration]] = {
+    val updatedRegistration = request.registration.organisationDetails.organisationType match {
       case Some(OrgType.UK_COMPANY) | Some(OrgType.OVERSEAS_COMPANY_UK_BRANCH) =>
         updateUkCompanyDetails(journeyId)
       case Some(OrgType.REGISTERED_SOCIETY) => updateRegisteredSocietyDetails(journeyId)
       case Some(OrgType.SOLE_TRADER)        => updateSoleTraderDetails(journeyId)
       case Some(OrgType.PARTNERSHIP)        => updatePartnershipDetails(journeyId)
       case _                                => throw new InternalServerException(s"Invalid organisation type")
     }
+    updatedRegistration
```

Comment:
Maybe drop the intermediate `updatedRegistration` val? It flows better if the flatMap occurs on the match result.

Reply:
Done